### PR TITLE
Fix state leak when a stopped stream is reset

### PR DESCRIPTION
**Pull request by Ralith**

### CODE REVIEW

**Review by djc — APPROVED** [review]

**Comment by djc** [comment]
Does this merit a bugfix release on the 0.6 branch?

**Comment by Ralith** [comment]
This has good odds of happening whenever an open receive stream is dropped, so that sounds sensible.

File: quinn-proto/src/connection/streams.rs [hunk 1]
```diff
@@ -350,7 +350,12 @@ impl Streams {
             return Ok(ShouldTransmit::new(false));
         }
         let bytes_read = rs.assembler.bytes_read();
-        self.on_stream_frame(true, id);
+        let stopped = rs.assembler.is_stopped();
+        if stopped {
```
Comment (Matthias247) [inline quinn-proto/src/connection/streams.rs:354]: Is just `stopped` enough to remove it, or should it be "dropped by application". One can call `stop` without dropping the stream. The next interaction would then lead to an unknown stream error?

Comment (Ralith) [reply quinn-proto/src/connection/streams.rs:354]: IMO this is sufficient. There's no operation that's meaningful to perform on a stopped stream. The documentation for each `pub fn stop` says: `Once stopped, a stream cannot be read from any further.` Perhaps we should make this clearer.

**Pull request merged**